Undo/COUndoTrackHistoryCompaction.h

```diff
@@ -0,0 +1,89 @@
+/**
+    Copyright (C) 2015 Quentin Mathe
+
+    Date:  June 2015
+    License:  MIT  (see COPYING)
+ */
+
+#import <Foundation/Foundation.h>
+#import <EtoileFoundation/EtoileFoundation.h>
+#import <CoreObject/COHistoryCompaction.h>
+
+@class COUndoTrack, COCommandGroup;
+
+/** 
+ * @group Undo
+ * @abstract A compaction strategy that targets the history located in the undo
+ * track tail or outside the undo track.
+ *
+ * @section Conceptual Model
+ *
+ * This strategy ensures that an undo track will continue to work flawlessly 
+ * for all operations (undo, redo, selective undo, set current node etc.), even 
+ * after compacting the history in the store.
+ *
+ * The undo track tail can be cut, but the remaing part up to the head will
+ * be kept intact.
+ *
+ * Revisions, persistent roots and branches referenced by other undo tracks or
+ * none are not protected by this strategy. This means you must be careful not 
+ * to break other undo tracks, when you are using multiple ones.
+ *
+ * @section Common Use Cases
+ *
+ * The most common use case is when you want to free space in a CoreObject store.
+ *
+ * Before passing the compaction strategy to -[COSQLiteStore compactHistory:], 
+ * you must always call -compute.
+ *
+ * For now, COUndoTrackHistoryCompaction is not thread-safe.
+ */
+@interface COUndoTrackHistoryCompaction : NSObject <COHistoryCompaction>
```

Comment:
Yes, the idea is that the undo track can keep anything it references alive:

- old revisions
- deleted persistent roots
- deleted branches

Non-deleted persistent roots and non-deleted branches won't be finalized, even when they are not referenced by an undo track.

In my latest application iteration, I only expose undo tracks. The history compaction is driven by a pattern track that match all tracks in the undo track store.

Previously I was exposing some current branches directly in the UI (CORevision presented as a COTrackNode), but I was still recording every commit on undo tracks, so the GC targeting the revisions was able to work properly without breaking the branch revision graphs at the UI level (e.g. no missing revisions in the middle of a branch history or no revisions unrelated to the undo track nodes kept alive).